control-plane-operator/controllers/hostedcontrolplane/pki/konnectivity.go

```diff
@@ -44,6 +45,8 @@ func ReconcileKonnectivityAgentSecret(secret, ca *corev1.Secret, ownerRef config
 func ReconcileKonnectivityWorkerAgentSecret(cm *corev1.ConfigMap, ca *corev1.Secret, ownerRef config.OwnerRef) error {
 	ownerRef.ApplyTo(cm)
 	secret := manifests.KonnectivityAgentSecret("kube-system")
+	// Ignore errors here, the configmap might be empty initially
+	yaml.Unmarshal([]byte(cm.Data[util.UserDataKey]), &secret)
```

Comment:
I think you want to pass `secret` instead of `&secret` since secret is already a pointer.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1f48cb2]

goroutine 468 [running]:
github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/pki.reconcileSignedCertWithKeysAndAddresses(0x0, 0xc000c7d2c0, 0x0, 0x261f6c3, 0x12, 0x2613ea8, 0xa, 0x5, 0x347b3e0, 0x1, ...)
	/hypershift/control-plane-operator/controllers/hostedcontrolplane/pki/cert.go:42 +0x172
github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/pki.reconcileSignedCertWithKeys(0x0, 0xc000c7d2c0, 0x0, 0x261f6c3, 0x12, 0x2613ea8, 0xa, 0x5, 0x347b3e0, 0x1, ...)
	/hypershift/control-plane-operator/controllers/hostedcontrolplane/pki/cert.go:29 +0x16f
github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/pki.reconcileSignedCert(0x0, 0xc000c7d2c0, 0x0, 0x261f6c3, 0x12, 0x2613ea8, 0xa, 0x5, 0x347b3e0, 0x1, ...)
	/hypershift/control-plane-operator/controllers/hostedcontrolplane/pki/cert.go:25 +0x10f
github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/pki.ReconcileKonnectivityAgentSecret(0x0, 0xc000c7d2c0, 0x0, 0x0, 0x0)
	/hypershift/control-plane-operator/controllers/hostedcontrolplane/pki/konnectivity.go:42 +0xb1
github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/pki.ReconcileKonnectivityWorkerAgentSecret(0xc000990140, 0xc000c7d2c0, 0xc000c67130, 0x0, 0x0)
	/hypershift/control-plane-operator/controllers/hostedcontrolplane/pki/konnectivity.go:50 +0x1ec
```